securetar/__init__.py

```diff
@@ -196,7 +196,18 @@ def __init__(
     def __enter__(self) -> tarfile.TarFile:
         """Start context manager tarfile."""
         tar_info = tarfile.TarInfo(name=str(self._name))
-        tar_info.mtime = int(time.time())
+        if self.outer_tar.format == tarfile.PAX_FORMAT:
+            # Ensure we always set mtime as a float to force
+            # a PAX header to be written.
+            #
+            # This is necessary to
+            # handle large files as TarInfo.to_buf will try to
+            # use a shorter ustar header if we do not have at
+            # least one float in the tarinfo.
+            # https://github.com/python/cpython/blob/53b84e772cac6e4a55cebf908d6bb9c48fe254dc/Lib/tarfile.py#L1066
+            tar_info.mtime = time.time()
```

Comment:
Nit: Would it maybe make more sense to fake a big size here to force PAX header? As this is in the end the reason we want the pax header... :thinking:

Reply:
I thought about doing that, but we need the size to be zero in case the inner tar creation raises an exception, so it can still extract an empty file

Reply:
Ah right, this is the inner tar file, which we don't modify later. Why is then PAX header even relevant? So its really just the fact if it is a PAX header which causes problems on extract? :thinking:

Reply:
Because there were no floats, the code in `to_buf` optimized away the pax header, when we later go back and rewrite the header we end up writing the pax header which is longer so it overwrites the data stream

Reply:
I mean it doesn't really matter all that much, but can't then go back later also in the exception case and write the size back to zero?

Reply:
Oh I see, that is not "us" as in securetar, in the inner tar case. Ok make sense.